app/main.py

```diff
@@ -2,5 +2,32 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    arguments = {}
+
+    def wrapper(*args) -> Callable:
+        if args not in arguments and isinstance(args, (int, float, tuple)):
+            result = func(*args)
+            arguments.update({args: result})
+            print("Calculating new result")
+            return result
+        print("Getting from cache")
+        return arguments[args]
+    return wrapper
+
+
+@cache
```

Comment:
I see that you teseted your code , but please remove it before creating PR :) 
Push only solution for task
( Remove all code from 18 to 33 line)